test/svid/jwtsvid/test_jwt_svid.py

```diff
@@ -172,6 +176,117 @@ def test_valid_parse_insecure(test_input_token, test_input_audience, expected):
 """
     parse_and_validate tests
 
-    TBD
-
 """
+
+
+@pytest.mark.parametrize(
+    'test_input_token,test_input_jwtBundle, test_input_audience, expected',
+    [
+        (
+            '',
+            None,
+            ['spire'],
+            INVALID_INPUT_ERROR.format('token cannot be empty'),
+        ),
+        (
+            'eyJhbGciOiJFUzI1NiIsImtpZCI6Imd1eTdsOWZSQzhkQW1IUmFtaFpQbktRa3lId2FHQzR0IiwidHlwIjoiSldUIn0.eyJhdWQiOlsib3RoZXItc2VydmljZSJdLCJleHAiOjE2MTIyOTAxODMsImlhdCI6MTYxMjI4OTg4Mywic3ViIjoic3hthrtmZlOi8vZXhhbXBsZS5vcmcvc2VydmljZSJ9.W7CLQvYVBQ8Zg3ELcuB1K9hE4I9wyCMB_8PJTZXbjnlMBcgd0VDbSm5OjoqcGQF975eaVl_AdkryJ_lzxsEQ4A',
+            None,
+            ['spire'],
+            INVALID_INPUT_ERROR.format('jwt_bundle cannot be empty'),
+        ),
+    ],
+)
+def test_invalid_parameters_parse_and_validate(
+    test_input_token, test_input_jwtBundle, test_input_audience, expected
+):
+    with pytest.raises(ValueError) as exception:
+        JwtSvid.parse_and_validate(
+            test_input_token, test_input_jwtBundle, test_input_audience
+        )
+    assert str(exception.value) == expected
+
+
+def test_valid_parse_and_validate_RSA():
+    rsa_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+    jwt_bundle = JwtBundle(TrustDomain('test.orgcom'), {'kid1': rsa_key.public_key()})
+    rsakeypem, _ = get_keys_pems(rsa_key)
+    token = jwt.encode(
+        {
+            'aud': ['spire', 'test', 'valid'],
+            'sub': 'spiffe://test.orgcom/',
+            'exp': timegm(
+                (
+                    datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+                ).utctimetuple()
+            ),
+        },
+        algorithm="RS256",
+        key=rsakeypem,
+        headers={'alg': 'RS256', 'typ': 'JWT', 'kid': 'kid1'},
+    )
+    JwtSvid.parse_and_validate(token, jwt_bundle, 'spire')
+    assert True
+
+
+def test_valid_parse_and_validate_EC():
+    ec_key = ec.generate_private_key(ec.SECP384R1(), default_backend())
+    jwt_bundle = JwtBundle(TrustDomain('test.orgcom'), {'kid_ec': ec_key.public_key()})
+    eckeypem, _ = get_keys_pems(ec_key)
+    token = jwt.encode(
+        {
+            'aud': ['spire', 'test', 'valid'],
+            'sub': 'spiffe://test.orgcom/',
+            'exp': timegm(
+                (
+                    datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+                ).utctimetuple()
+            ),
+        },
+        algorithm="ES256",
+        key=eckeypem,
+        headers={'alg': 'ES256', 'typ': 'JOSE', 'kid': 'kid_ec'},
+    )
+    JwtSvid.parse_and_validate(token, jwt_bundle, 'spire')
+    assert True
+
```

Comment:
A couple test that I feel are missing: 
- what happens if the kid of the token is not found in the authorities dict when trying to parse the svid
- what happens if the authorities is found but the token doesn't pass the validation using the authority (a kid - public key mismatch)

Reply:
Good ideas.